### PR TITLE
feat: endBlock and initChain handlers

### PR DESCRIPTION
**Pull request by Alex-Werner**
## Issue being fixed or feature implemented

This PR introduce two new handlers `endBlockHandler` and `initChainHandler`. 
In the initial implementation, those handlers get the lastestCoreChainLock passed to the response.

## What was done?
- feat: created initChainHandlerFactory
- feat: created endBlockHandlerFactory
- feat: passed lastestCoreChainLock to the handler

## How Has This Been Tested?
- Added additional tests

## Breaking Changes
N/A

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [X] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by Alex-Werner** [comment]
Requesting your eyes on this @shumkov for some guidance on expectation. 
Do we want it passed as a ChainLock primitives (Dashcore-lib) or stringified ?

File: lib/abci/handlers/endBlockHandlerFactory.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  blockchainState,
+  logger,
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestEndBlock} request
+   * @return {Promise<abci.ResponseEndBlock>}
+   */
+  async function endBlockHandler({ latestCoreChainLock }) {
```
Comment (shumkov) [inline lib/abci/handlers/endBlockHandlerFactory.js:25]: We need a singleton class LatestCoreChainLock to store ChainLock primitive that should be passed here as a dependency (as an endBlockHandlerFactory argument)

File: lib/abci/handlers/initChainHandlerFactory.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestInitChain} request
+   * @return {Promise<abci.ResponseInitChain>}
+   */
+  async function initChainHandler({ latestCoreChainLock }) {
+    logger.info('Init chain');
+    return new ResponseInitChain({
```
Comment (shumkov) [inline lib/abci/handlers/initChainHandlerFactory.js:27]: It requires appHash https://docs.tendermint.com/master/spec/abci/abci.html#initchain

We should pass it from blockchainState.

**Comment by shumkov** [comment]
@Alex-Werner I guess we need to serialize it and pass as a Buffer.

**Comment by Alex-Werner** [comment]
Replaced by #434

**Pull request closed without merging**